components/eventing-controller/pkg/handlers/beb.go

```diff
@@ -238,7 +272,7 @@ func (b *Beb) getSubscription(name string) (*types.Subscription, error) {
 		return nil, fmt.Errorf("get subscription failed: %v", err)
 	}
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("get subscription failed: %v; %v", resp.StatusCode, resp.Message)
+		return nil, fmt.Errorf("get subscription failed: %w; %v", errors.New(strconv.Itoa(resp.StatusCode)), resp.Message)
```

Comment:
Instead of doing the error.New twice, how about we keep this as is and create a 404 error constant to compare in [SyncSubscription:123](https://github.com/kyma-project/kyma/pull/11807/files#diff-17d39986bafe1b1562ce9db3c9fb4828533f6d1badfa43fc5476892b27dc88d1R123)

Reply:
The idea is to move from the old error handling based on string compare with the new one, exposed by the new "errors" package coming with Go 1.13 (or 1.14), I'm not sure. This will allow to have a deeper errors stack if calls are chained.